src/components/CreateMediaFolderModal/CreateMediaFolderModal.tsx

```diff
@@ -0,0 +1,305 @@
+import {
+  Box,
+  FormControl,
+  HStack,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalFooter,
+  ModalHeader,
+  ModalOverlay,
+  SimpleGrid,
+  Text,
+} from "@chakra-ui/react"
+import { yupResolver } from "@hookform/resolvers/yup"
+import {
+  Button,
+  FormErrorMessage,
+  FormLabel,
+  Input,
+  ModalCloseButton,
+} from "@opengovsg/design-system-react"
+import { AxiosError } from "axios"
+import _ from "lodash"
+import { useEffect } from "react"
+import { FormProvider, useForm } from "react-hook-form"
+import { UseMutateAsyncFunction } from "react-query"
+
+import { DirectorySettingsSchema } from "components/DirectorySettingsModal"
+import { ImagePreviewCard } from "components/ImagePreviewCard"
+
+import { getSelectedMediaDto } from "utils/media"
+
+import { GetMediaSubdirectoriesDto, MediaData } from "types/directory"
+import { MiddlewareError } from "types/error"
+import {
+  MediaFolderCreationInfo,
+  MediaLabels,
+  SelectedMediaDto,
+} from "types/media"
+
+interface CreateMediaFolderModalProps {
+  originalSelectedMedia: SelectedMediaDto[]
+  mediaLabels: MediaLabels
+  subDirectories: GetMediaSubdirectoriesDto | undefined
+  mediaData: (MediaData | undefined)[]
+  isWriteDisabled: boolean | undefined
+  isOpen: boolean
+  isLoading: boolean
+  onClose: () => void
+  onProceed: UseMutateAsyncFunction<
+    void,
+    AxiosError<MiddlewareError>,
+    MediaFolderCreationInfo,
+    unknown
+  >
+}
+
+const getButtonLabel = (
+  originalSelectedMedia: SelectedMediaDto[],
+  selectedMedia: SelectedMediaDto[],
+  mediaLabels: MediaLabels
+) => {
+  const {
+    singularDirectoryLabel,
+    singularMediaLabel,
+    pluralMediaLabel,
+  } = mediaLabels
+
+  if (originalSelectedMedia.length > 0) {
+    return `Create ${singularDirectoryLabel}`
+  }
+
+  if (selectedMedia.length > 1) {
+    return `Add ${selectedMedia.length} ${pluralMediaLabel} to ${singularDirectoryLabel}`
+  }
+
+  if (selectedMedia.length === 1) {
+    return `Add ${singularMediaLabel} to ${singularDirectoryLabel}`
+  }
+
+  return `Add ${pluralMediaLabel} to ${singularDirectoryLabel}`
+}
+
+export const CreateMediaFolderModal = ({
+  originalSelectedMedia,
+  mediaLabels,
+  subDirectories,
+  mediaData,
+  isWriteDisabled,
+  isOpen,
+  isLoading,
+  onClose,
+  onProceed,
+}: CreateMediaFolderModalProps): JSX.Element => {
+  const {
+    singularMediaLabel,
+    pluralMediaLabel,
+    singularDirectoryLabel,
+    pluralDirectoryLabel,
+  } = mediaLabels
+
+  const existingTitles = subDirectories?.directories.map(
+    (directory) => directory.name
+  )
+
+  const methods = useForm<MediaFolderCreationInfo>({
+    mode: "onChange",
```

Comment:
Checking with design but for the time being, I think I will standardise to use `onTouched` since there is a performance impact when using `onChange`. Fixed in dfd807ca.

Reply:
Ok design says to use `onTouched`, so this is changed.